src/config.py

```
@@ -0,0 +1,121 @@
+                return {"endpoint": self.endpoint, "key": self.key}
+        except Exception as e:
+            logger.error(f"Error getting connection parameters: {e}")
+            return {"endpoint": "https://localhost:8081", "key": self.key}
+
+    def validate_config(self) -> bool:
+        """Validate the configuration is usable."""
+        try:
+            if self.connection_string:
+                return bool(self.connection_string.strip())
+            return bool(self.endpoint and self.key)
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)